pennylane_lightning/core/src/utils/SharedLibLoader.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file
+ * Dynamic shared library functions API wrapper.
+ */
+#pragma once
+#include <string>
+
+#if defined(__APPLE__) || defined(__linux__)
+#include <dlfcn.h>
+#define PL_DLOPEN(NAME, ARG) dlopen(NAME, ARG)
+#define PL_DLERROR() dlerror()
+#define PL_DLCLOSE(NAME) dlclose(NAME)
+#define PL_DLSYS(NAME, SYMBOL) dlsym(NAME, SYMBOL)
+#endif
```

Comment:
@multiphaseCFD added these macros to simplify the code. would you think that we can also update PR #666?